tests/functional/nccl_message_transfer.c

```diff
@@ -141,204 +143,232 @@ int main(int argc, char* argv[])
 		support_gdr[dev] = is_gdr_supported_nic(props.ptrSupport);
 	}
 
-	/* Test all devices */
-	for (int dev_idx = 0; dev_idx < ndev; dev_idx++) {
+	for (size_t szidx = 0; szidx < sizeof(send_sizes) / sizeof(send_sizes[0]); szidx++) {
```

Comment:
Having this at the outer loop seems excessive. Couldn't we do all the initialization, connection establishment things, and then iterate over the sizes?

Reply:
good point, I am fixing it.